### PR TITLE
Wfappoint 25339 abstract notify&update appointment action tasks no longer works with wf v6

### PR DESCRIPTION
**Pull request by ManssourBlh**
https://dev.lutece.paris.fr/bugtracker/issues/25339 
https://dev.lutece.paris.fr/bugtracker/issues/25457

### CODE REVIEW

File: src/java/fr/paris/lutece/plugins/workflow/modules/appointment/web/UpdateAppointmentCancelActionTaskComponent.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public String getDisplayConfigForm( HttpServletRequest request, Locale locale, I
 
         List<Action> listActions = _actionService.getListActionByFilter( filter );
 
-        if ( action.getStateAfter( ).getId( ) == action.getStateBefore( ).getId( ) )
+        if ( action.getStateAfter( ).getId( ) == action.getListIdStateBefore().get(0) )
```
Comment (ryahiaoui) [inline src/java/fr/paris/lutece/plugins/workflow/modules/appointment/web/UpdateAppointmentCancelActionTaskComponent.java:109]: Why is the comparison made only on the first state in the list?

**Review by ryahiaoui — COMMENTED** [review]
Why is the comparison made only on the first state in the list?

File: src/java/fr/paris/lutece/plugins/workflow/modules/appointment/web/UpdateAppointmentCancelActionTaskComponent.java (hunk 1 above)
Comment (ManssourBlh) [reply src/java/fr/paris/lutece/plugins/workflow/modules/appointment/web/UpdateAppointmentCancelActionTaskComponent.java]: The revised code now properly checks if `action.getStateAfter().getId()` is present in any of the states in the `listIdStateBefore`. This modification ensures that all possible states of departure are considered
Thank you for your feedback

**Comment by ryahiaoui** [comment]
Thanks!
Could you please upgrade the plugin-workflow dependency to version 6.0.0? Additionally, could you please consider the latest update regarding the "_alternativeStateAfter" property of the Action class in the workflow?

 <dependency>
            <groupId>fr.paris.lutece.plugins</groupId>
            <artifactId>plugin-workflow</artifactId>
            <version>[6.0.0,)</version>
            <type>lutece-plugin</type>
   </dependency>

**Pull request merged**